roles/openshift_router/tasks/main.yml

```diff
@@ -0,0 +1,11 @@
+---
+
+# TODO: Centralized openshift_config openshift_facts??
+# TODO: anchor this on a master
```

Comment:
You are correct an infrastructure node is a better answer.  bin/cluster and playbooks will need to be updated to create the correct number of node types and set the selectors.